lib/realtime/api.ex

```diff
@@ -112,23 +124,15 @@ defmodule Realtime.Api do
 
   """
   def update_tenant(%Tenant{} = tenant, attrs) do
-    tenant
-    |> Tenant.changeset(attrs)
-    |> tap(&maybe_trigger_disconnect/1)
-    |> Repo.update()
-  end
+    changeset = Tenant.changeset(tenant, attrs)
+    updated = Repo.update(changeset)
 
-  defp maybe_trigger_disconnect(%Ecto.Changeset{
-         changes: changes,
-         valid?: true,
-         data: %{external_id: external_id}
-       })
-       when is_map_key(changes, :jwt_jwks) or is_map_key(changes, :jwt_secret) do
-    Phoenix.PubSub.broadcast!(Realtime.PubSub, "realtime:operations:" <> external_id, :disconnect)
+    maybe_invalidate_cache(changeset)
```

Comment:
Is there a race condition here?

What if the invalidation takes a long time and the DB reconnects before the Tenant cache has been discarded 🤔 This way it might still connect to the database with the old  settings?

Reply:
I have no clue honestly ... we might just need to trust cachex? 😅